webchat/utils/chat.js

```diff
@@ -0,0 +1,202 @@
+'use strict';
+(function () {
+  const app = angular.module('webchat');
+
+  app.factory('Chat', ['ChatMessage', (ChatMessage) => {
+    const Chat = class Chat {
+      /**
+       * Creates a new Chat object, which will handle a webrtc peer connection,
+       * its events and video/audio/data channels functionality.
+       * @param {MediaStream} stream - video/audio stream object
+       * @param {string} selfId - firebase id of the current user
+       */
+      constructor(stream, selfId) {
```

Comment:
Create constructor when no MediaStream is passed in. It should call the commom constructor with a "black silence" media stream.